resources/model/OpenStudio.idd

```diff
@@ -31186,13 +31188,100 @@ OS:ZoneHVAC:CoolingPanel:RadiantConvective:Water,
        \minimum 0
        \maximum 1
 
+OS:ZoneHVAC:EvaporativeCoolerUnit,
+       \memo Zone evaporative cooler. Forced-convection cooling-only unit with supply fan,
+       \memo 100% outdoor air supply. Optional relief exhaust node
+       \min-fields 17
+  A1,  \field Handle
+       \type handle
+       \required-field
+  A2,  \field Name
+       \required-field
+       \type alpha
+       \reference ConnectionObject
+  A3,  \field Availability Schedule Name
+       \note Availability schedule name for this system. Schedule value > 0 means the system is available.
+       \note If this field is blank, the system is always available.
+       \required-field
+       \type object-list
+       \object-list ScheduleNames
+  A4,  \field Availability Manager List Name
+       \note Enter the name of an AvailabilityManagerAssignmentList object.
+       \type object-list
+       \object-list SystemAvailabilityManagerLists
+  A5,  \field Outdoor Air Inlet Node Name
+       \required-field
+       \type object-list
+       \object-list ConnectionNames
+       \note this is an outdoor air node
+  A6,  \field Cooler Outlet Node Name
+       \required-field
+       \type object-list
+       \object-list ConnectionNames
+       \note this is a zone inlet node
+  A7,  \field Zone Relief Air Node Name
+       \type object-list
+       \object-list ConnectionNames
+       \note this is a zone exhaust node, optional if flow is being balanced elsewhere
```

Comment:
Ok, I think the node connections are wrong at model namespace.

You've set inletPort to be "Outdoor Air Inlet Node Name", but that should be an OA node. And I don't think it should be in the IDD.

inletPort should be Zone Relief Air Node Name, so that the addToThermalZone works

> Zone Relief Air Node Name : This optional alpha input field defines the name of an HVAC system node which can extract air from the zone to balance the air supplied to the zone by the cooler outlet node. This node name would match the name of a zone exhaust air node.

. But we should also have a Boolean field like "Flow Is Balanced" to determine whether we write it to IDF or not (or maybe we just always assume it is balanced here, not elsewhere)

Reply:
Proposed a change in 373c3943bee6cc6d5f9b4df57f98371f8ff694fb

Reply:
That comment stands though:

> But we should also have a Boolean field like "Flow Is Balanced" to determine whether we write it to IDF or not (or maybe we just always assume it is balanced here, not elsewhere)